### ISSUE TITLE
drupal:sync command doesn't allow to specify options for drush sql:sanitize command

### ISSUE THREAD

**Issue by miloskroulik** [acquia/blt#3138]
`drupal:sync` command [calls](https://github.com/acquia/blt/blob/03e26f02a789d3deb5b97b69c381594985c3e234/src/Robo/Commands/Sync/SyncCommand.php) `drush sql-sanitize` command and doesn't allow for specifying options. 

This means that I can't disable sanitizing emails, if I need. It's important for me because it's possible to implement a `post-command` defining additional sanitization, but not override default one easily. I would like to avoid having maintaining my own version of the `drupal:sync` command.

**Comment by dalibor-matura** [comment]
To add a bit of information to what Milos wrote above: We have a problem to whitelist accounts/emails (and possibly other data) that we do not want to sanitized (there should stay unchanged).

So basically "whitelisting" is not possible with current structure of `blt drupal:sync` command. We've given it some thoughts and the only solution for us would be to rewrite the `blt drupal:sync` command and patch or create our own. This is not a good solution and it would bring significatn technical debt.

Can you help us by:
- providing instructions how to solve the whitelisting in a clean way (blt way)
- or update `blt drupal:sync` command to accept `drush sql-sanitize` options
- or provide a different clean solution

?

Thanks,
Dalibor

**Comment by lpeabody** [comment]
This is more of a "how to use Drush 9" question rather than a BLT question. Passing options to Drush commands is handled by Drush's configuration loading mechanism. So, the decision making tree is:

1) Do we want BLT to run sql:sanitize? 
    * No -> set BLT's `drush.sanitize` configuration value to `false`.
    * Yes -> don't do anything more with BLT
2) If Yes, configure Drush options for the sql:sanitize command:

From example.drush.yml, lines 11-21:

```
# Directories and Discovery
#
# Rename this file to drush.yml and copy it to one of the places listed below
# in order of precedence:
#
# 1.  Drupal site folder (e.g. sites/{default|example.com}/drush.yml).
# 2.  Drupal /drush and sites/all/drush folders, or the /drush folder
#       in the directory above the Drupal root.
# 3.  In any location, as specified by the --config (-c) option.
# 4.  User's .drush folder (i.e. ~/.drush/drush.yml).
# 5.  System wide configuration folder (e.g. /etc/drush/drush.yml or C:\ProgramData\Drush\drush.yml).
```

So, in drush.yml in one of those locations set the following:

```
command:
  sql:
    sanitize:
      options:
        sanitize-password: 'no'
        sanitize-email: 'no'
        whitelist-fields:
          - field_a
          - field_b
```

You don't have to set all of those options for the sql:sanitize command, but this is how you would do that.

IMHO, this is the BLT-way and Drush-way of doing things. You determine if you want BLT to run the sanitize command, and then further customize the sanitize command via Drush configuration.

**Comment by nmillin** [comment]
I know this is an old issue, but the whitelist-fields wasn't working for me so I wanted to let others know in case they find this issue. **Note** This might have changed between when lpeabody commented on this or how drush 10 works. 

The sanitize() in SanitizeUserFieldsCommands.php (drush v10.3.6) is expecting whitelist-fields to be a string and not an array. Also, whitelist-fields is deprecated and allowlist-fields should be used instead.

TL;DR use the following and you'll have a good time:
`command:
  sql:
    sanitize:
      options:
        sanitize-password: 'no'
        sanitize-email: 'no'
        allowlist-fields: 'field_a,field_b'`
